### DISCUSSION TITLE
Hardware-Entscheidung: HM-MOD-RPI oder RFUSB?

### DISCUSSION THREAD

**Discussion by Micha-DU** [OpenCCU/OpenCCU#2148] [🇩🇪 Fragen&Antworten]
Hallo zusammen,

ich benutze derzeit Raspberrymatic auf einem Pi 3B in Verbindung mit einem RF-USB. Dieser "zickt" leider von Beginn an (wird meist bei einem Neustart nicht erkannt, scheinbar ein Verbindungsproblem im Bereich der Kontaktbahnen des angelöteten USB-Steckers) und daher möchte ich ihn nun lieber ersetzen. In diesem Zusammenhang würde ich eigentlich gern auf die platzsparende und kostengünstige Lösung mit dem "Huckepack"-Modul HM-MOD-RPI (-PCB?) umsteigen, bin nun aber in verschiedenen Beiträgen über Aussagen gestolpert, die auf einen eingeschränkten Funktionsumfang hinweisen und eher zu einem RF-USB raten.

Gibt es da tatsächlich gravierende Unterschiede und warum wird dann teilweise auf die deutlich teurere Variante mit dem RPI-RF-MOD hingewiesen?

Ich möchte gern auf eine Lösung unter Benutzung von "HomeAssistant" umsteigen und bin nun total unsicher, welche Hardware "notwendig" ist.

Kann da vielleicht jemand "Aufklärungsarbeit" leisten? ;-)

Vielen Dank vorab und VG
Micha

**Comment by Baxxy13** [comment]
Nimm das RPI-RF-MOD.
Es steckt auch in der CCU3 und wird daher von eQ-3 noch "gepflegt", also bspw. bei Bedarf mit neuer Firmware versorgt.
Außerdem kann nur dieses den CARRIER_SENSE anzeigen, man kann einen HAP als LAN-Router einbinden und HmIP-Wired nutzen.
All das geht mit dem HM-MOD-RPI-PCB nicht, mit dem HmIP-RFUSB hingegen schon.

HAP und DRAP funktionieren aktuell nicht wenn RaspberryMatic als HA-AddOn läuft, egal welches "Funkmodul" genutzt wird.

Pi3B + RPI-RF-MOD passen prima in das ELV "Charly" Gehäuse und natürlich in ein original CCU3-Gehäuse.

**Reply by Micha-DU** [reply]
Danke für die schnelle Antwort.

Ich möchte eigentlich eine eher "kleine" Installation, deswegen würde mich ein Umbau in das Charly-Gehäuse eher abschrecken. Andererseits... ich habs mir gerade mal angesehen... das schwarze sieht ja schon irgendwie "schick" aus und man hat dann alles in einem... kompakt und "verstaut"... hmmmm.

Ich hatte tatsächlich vor, eine HA-AddOn-Installation zu testen... in der Hoffnung, damit dann auch meine restlichen ("nicht-HmIP") Komponenten inkl. Sprachsteuerung über Alexa einbinden zu können. Wenn ich mir nicht zu viel vorgenommen habe - denn bei iobroker bin ich z. B. hoffnungslos gescheitert ;-)

"Carrier_Sense" zeigt mein RF-USB auch an, aber wofür ist diese Anzeige überhaupt gut? Die Funktion "LAN-Router" benötige ich nicht, ebensowenig HmIP-Wired (das meintest Du mit "DRAP"?). Spricht eigentlich eher für die "kleine" Lösung mit dem HM-MOD-RPI. Einzig die Pflege der FW würde mich ein wenig unruhig machen... wobei natürlich auch der Preis eine Rolle spielt. Ich kann gerade einen HM-MOD-RPI für 15€ bekommen, die RPI-RF-MODs liegen eher bei 40€ plus 15€ Charly-Gehäuse plus Versand... 

Da bin ich immer noch "unschlüssig". Wichtig ist, ob es - bis auf die oben genannten, aber auch nicht benötigten Funktionen - "generell" mit dem kleinen Modul funktioniert.

VG Micha

**Reply by Baxxy13** [reply]
Wenn einem die genannten Punkte unwichtig sind kann man natürlich auch das kleine Funkmodul nehmen.

Der CARRIER_SENSE ist immer dann interessant wenn man Probleme mit IP-Geräten hat. (beim Anlernen / Komm-Störungen usw.)
Er zeigt "Störpegel" auf der von HmIP genutzten Funkfrequenz an.

Wenn du sparen willst bist du bei "Smart-Home" an der falschen Stelle.

HM / HmIP Aktoren / Sensoren kosten kosten im Schnitt 60€ und du willst beim "Hauptgerät" dem Funkmodul knausern?
Aber gut, jeder wie er mag.

**Reply by jp112sdl** [reply]
Ich streue mal ein bisschen Salz in die Carrier-Sense-Wunde und würde unter Michas Voraussetzungen (kein HmIPW), nach all den Querelen mit dem CS in letzter Zeit (im Homematic Forum), das HM-MOD-RPI-PCB benutzen.

Das hat kein LBT/CCA und sendet auch noch, wenn ein Störpegel über der von eQ-3 festgelegten Schwelle anliegt.

**Reply by MichaelN0815** [reply]
Das ist ja ein interessanter Gesichtspunkt. D.h. Das kann nicht nur keinen CS anzeigen, es berücksichtigt den auch überhaupt nicht?

**Reply by jp112sdl** [reply]
Korrekt. Senden geht dadurch immer und empfangen je nachdem wie stark sich das Nutzsignal vom Störsignal absetzt (Signal-Rausch-Verhältnis).

Homematic IP (Funk) gibt es ja schon wesentlich länger als das RPI-RF-MOD und seine Geschwister.

Schau mal im Forum, wie viele HmIP-(Anlern)Probleme es früher mit dem HM-MOD-RPI-PCB gibt/gab.

**Reply by jens-maus** [reply]
Wäre das dann ggf. eine Diskussion mit dem Firmwareentwickler des RPI-RF-MOD/HmIP-RF-USB wert das hier ggf. andere Schwellen notwendig wären?!?

**Reply by MichaelN0815** [reply]
Wann stellt das Modul denn das Senden ein? Bei 100% CS? Wenn man bedenkt das viele schon bei 10% keine Geräte mehr anlernen können, scheint die Schwelle eher nicht zu niedrig angesetzt zu sein.

**Reply by jp112sdl** [reply]
> Wäre das dann ggf. eine Diskussion mit dem Firmwareentwickler des RPI-RF-MOD/HmIP-RF-USB wert das hier ggf. andere Schwellen notwendig wären?!?

Eventuell... aber das befindet sich aktuell noch im Hintergrund in Prüfung und wird noch etwas dauern. Ich will auch nix vorweg nehmen, da das nicht von mir abhängt.

> Wann stellt das Modul denn das Senden ein? Bei 100% CS? Wenn man bedenkt das viele schon bei 10% keine Geräte mehr anlernen können, scheint die Schwelle eher nicht zu niedrig angesetzt zu sein.

Es gibt 2 Möglichkeiten, festzustellen, ob der "Kanal belegt" ist.
 1. in Hardware - der Funkmodulchip wird mit einem Carrier Sense RSSI Schwellwert initialisiert.
  - bei Überschreiten einer fixen oder auch relativen Schwelle wird der Sender gar nicht erst aktiv
  - https://www.ti.com/lit/ds/symlink/cc1101.pdf S.46 ff
  - https://www.ti.com/lit/ug/swcu117i/swcu117i.pdf S.1638
  - der boolsche Status lässt sich abfragen
 2. in Software - der an den Funkmodulchip angeschlossene µC (beim RPI-RF-MOD im CC1310 integriert, beim HmIP-RFUSB ein EFM32) pollt zyklisch den RSSI Wert, nimmt eine eigene (vom Entwickler ausgedachte) Bewertung vor und gibt einen daraus berechneten Carrier Sense Wert aus.

Wie sich nun die %-Anzeige zusammensetzt, kann ich dir im Moment noch nicht sagen.
Vielleicht ist es 1. und es wird JA/NEIN über eine ZEIT gemessen.
Oder es ist 2. und es wird evtl. in einem engen Rahmen, z.B. -100 ... -90 dBm, eine Range von 0...100% CS gelegt.
Oder sogar eine Kombination aus 1. und 2.

**Reply by jens-maus** [reply]
> > Wäre das dann ggf. eine Diskussion mit dem Firmwareentwickler des RPI-RF-MOD/HmIP-RF-USB wert das hier ggf. andere Schwellen notwendig wären?!?
> 
> Eventuell... aber das befindet sich aktuell noch im Hintergrund in Prüfung und wird noch etwas dauern. Ich will auch nix vorweg nehmen, da das nicht von mir abhängt.

Ah ok, du bist also schon in entsprechendem Kontakt. Dann ist ja alles gut. Dachte ich könnte hier auch meine eQ3 Kontakte ggf. antriggern um hier ggf. den Firmware Autor sich dem ganzen mal etwas annehmen zu lassen. Vielleicht gibt es bzgl. CS Berechnung bzw. Schwellwerteinstellung bis das Funkmodul das senden einstellen noch Optimierungsbedarf.

**Reply by jp112sdl** [reply]
Das kannst du gern tun. Die Community hat die Kontakte nicht und muss sich mühsam durchboxen ;)

**Reply by jp112sdl** [reply]
@stan23 war so nett und hat mal die SPI-Kommunikation bei der Initialisierung des Funkmoduls mitgeschnitten und dekodiert.

Demnach wird beim HmIP-RFUSB der Carrier Sense Level auf -91.5 dBm festgelegt.
Die Signalisierung, ob CS über-/unterschritten, erfolgt mithilfe eines GPIO Pins.
Ein RSSI Polling findet nicht statt.

Insofern muss es wohl Annahme 1.) https://github.com/jens-maus/RaspberryMatic/discussions/2148#discussioncomment-4644817 sein.

**Reply by stan23** [reply]
Ich habe auch die Gegenprobe gemacht:
GDO2 an einem HmIP-RFMOD mit einem Drähtchen gegen VCC halten, und in einer OVA ohne Geräte geht der CS innerhalb etwa einer Minute auf 100%.
Kurzes Tasten gegen VCC erzeugt Werte im zweistelligen Prozentbereich.

**Reply by jens-maus** [reply]
Und was lernen wir jetzt daraus für eine möglich Verbesserung? Versteh nich nicht ganz was nun die Konsequenz daraus ist und ob es hier etwas gibt das ich eQ3 sagen könnte damit die was ändern könnten.

**Reply by stan23** [reply]
Das weiß ich auch nicht :)
Ich wollte nur bestätigen dass der CS beim HmIP-RFMOD tatsächlich nur über den Pin GDO2 erkannt wird, und dieser nur die Überschreitung der festen Schwelle erkennen kann.

**Reply by jens-maus** [reply]
D.h. Man könnte diesen Pin kappen und dann würde er trotzdem noch funken? Müsste man mal versuchen die Schwelle herauszubekommen bei der wirklich kein zuverlässiger Funkverkehr mehr möglich ist bei HmIP. Könnte mir vorstellen hier hat eQ3 ggf nicht das Optimum gewählt.

**Reply by jp112sdl** [reply]
Mir ging es in erster Linie darum, herauszufinden, ab welcher RSSI-Schwelle eQ-3 hier das Senden verbietet.

Interessant wäre es, den Wert auch beim RPI-RF-MOD ausfindig zu machen. Aber das ist unmöglich, weil µC und RF-Transceiver in einem Gehäuse stecken.

Über die Festlegung von **-91.5 dBm** als Grenzwert für den Carrier Sense kann man natürlich diskutieren.
Als Kanal-Belegt-Referenz, um Kollisionen mit anderen Homematic-Geräten zu vermeiden, ist der Carrier-Sense aus unserer Sicht unbrauchbar, da die Homematic(IP)-Pakete für das Funkmodul als valide Telegramme erkannt werden.

Der CS ist dann also nur ein Maß für das "Grundrauschen".

Das ist ja schön und gut, warum dann aber gar nicht mehr gesendet wird, ist mir unklar.

Wie dem auch sei - mein CS daheim liegt meist bei < 2% und hat stündliche Peaks von 10%. 
<img width="972" alt="Bildschirm­foto 2023-01-22 um 10 14 44" src="https://user-images.githubusercontent.com/18190139/213908491-c6271f25-5db2-4ab1-b1b8-7c633879b00e.png">

Laut AskSinAnalyzer sehe ich dann auch kurz ein Noise Level von -88dBm.

So ziemlich alle meine Geräte haben laut DevConfig-RSSI-Übersicht Werte von -80dBm oder besser.

Für das HmIP-RFUSB bzw. RPI-RF-MOD Funkmodul sind -88dBm jedoch ein klares Zeichen für "Oh, Träger erkannt, ich darf nicht senden." 
Für den Kunden nervig. Obwohl das Gerät technisch erreichbar wäre.

**Reply by MichaelN0815** [reply]
Das bedeutet zumindest
ein Löt Fehler kann zu 100% CS führen
ob  -91.5 dBm der richtige Grenzwert ist kann man nicht experimentell ermitteln, da das Funkmodul dann vermutlich gar nicht mehr sendet.

**Reply by jp112sdl** [reply]
Nachtrag, da sich mein Beitrag mit Jens' überschnitten hat:

> Müsste man mal versuchen die Schwelle herauszubekommen bei der wirklich kein zuverlässiger Funkverkehr mehr möglich ist bei HmIP.

Das ist halt schon individuell und hängt immer stark vom Signal-Rausch-Abstand ab.
Ich halte nur ein "Es geht dann das Gerät einfach nicht" sinnvoller als grundsätzlich ab einem festen Pegel zu sagen "Egal, ob es ginge. Wir machen es einfach nicht".

> Könnte mir vorstellen hier hat eQ3 ggf nicht das Optimum gewählt.

Ja, im EMV-Labor sieht die Welt sicher ganz anders aus, als in der wilden Welt mit vielen unterschiedlichen Umwelteinflüssen

**Reply by jp112sdl** [reply]
> Das bedeutet zumindest
ein Löt Fehler kann zu 100% CS führen

Ja, kann es.

> ob -91.5 dBm der richtige Grenzwert ist kann man nicht experimentell ermitteln, da das Funkmodul dann vermutlich gar nicht mehr sendet.

Nur weil lokal am Sender (Zentrale) ein starkes Rauschen von > 91.5dBm empfangen wird, heißt das ja noch gar nicht, ob auch beim Empfänger dieses Störsignal anliegt.

**Reply by stan23** [reply]
> Das bedeutet zumindest ein Löt Fehler kann zu 100% CS führen ob -91.5 dBm der richtige Grenzwert ist kann man nicht experimentell ermitteln, da das Funkmodul dann vermutlich gar nicht mehr sendet.

Ein Lötfehler an jedem der 7 anderen Pins führt auch dazu dass das Funkmodul nicht mehr sendet.
Der GDO2 verzeiht hier sogar mehr als die anderen, weil eine Unterbrechung (je nachdem wohin der Input dann driftet) oder ein Kurzschluss nach Masse dazu führt dass gar kein CS gemessen wird.

Ob die HmIP-RFUSB prüft, ob vor jedem empfangenen Packet mit entsprechender Empfangsstärke tatsächlich der GDO2 die Kanalbelegung erkannt hat, oder dann den GDO2 als defekt markiert und eine Ersatzbehandlung durchführt, wissen wir nicht. Vermutlich nicht :-)

**Reply by jp112sdl** [reply]
> ein Kurzschluss nach Masse dazu führt dass gar kein CS gemessen wird.

Na das wäre ja eigentlich eine praktikable Möglichkeit, die Carrier Sense Prüfung zu umgehen?

**Reply by MichaelN0815** [reply]
Eleganter wäre aber doch die Prüfung praxistauglicher zu gestalten. Auch wenn dazu Jens noch ein paar dicke Bretter bei eq3 bphren muss.
Besser ich habe eine Anzeige für die "Funkhygiene", als gar keine. Besser wäre natürlich das Funkmodul versucht es trotzdem bzw. Stellt das Senden erst deutlich später ein. 
Was ich noch nicht so ganz nachvollziehen kann - es gibt ja Berichte aus dem Forum, wo Leute bei 10% CS kein Gerät mehr anlernen konnten. 
Wenn in einer Minute der CS auf 100% steigt, sind 10% 6 Sekunden in denen das Funkmodul nicht sendet?! 
Und das reicht um das anlernen zu topedieren?

**Reply by jp112sdl** [reply]
Ich find es nur noch nervig, wie oft im Forum HmIP-Probleme im Zusammenhang mit dem Carrier Sense auftreten.
Ich kann mich nicht daran erinnern, dass vor der Erfindung des Features so extrem häufig von HmIP-Kommunikationsproblemen berichtet wurde.

**Reply by Baxxy13** [reply]
So ist auch mein Empfinden.
Seitdem @hobbyquaker das damals "enthüllt" hatte gingen die "Carrier-Sense Probleme" im Forum los.
Die Frage wäre wie er drauf gekommen ist. 
Und auch ob es das "Feature" schon immer gab (vielleicht mit tolreanterer Schwelle oder gar inaktiv), oder ob es mit irgendeiner Funkmodul-Firmware erst aktiviert wurde.

**Reply by jens-maus** [reply]
> Ich find es nur noch nervig, wie oft im Forum HmIP-Probleme im Zusammenhang mit dem Carrier Sense auftreten. Ich kann mich nicht daran erinnern, dass vor der Erfindung des Features so extrem häufig von HmIP-Kommunikationsproblemen berichtet wurde.

Also ich meine mich zu erinnern das vor der Einführung des CarrierSense HmIP Geräte durchaus öfter als Gerätekommunikation gestört aufgelistet bzw. angemeckert wurden im Forum. Aber durch den CarrierSense schiebt man das nun eben unmittelbar darauf und sucht sich nicht einen Wolf mehr woher es kommt das das eine Gerät vllt. noch erreichbar ist und das andere nicht. Deshalb finde ich das prinzipiell schon OK und ich selbst hab damit ja auch überhaupt keine Probleme. Aber ich denke trotzdem das es hier ggf. noch gewisses optimierungspotential gibt was die Schwelle des CS angeht. Wenn ihr also konkrete Vorschläge für verbessungen habt (weil ihr die HW-Geeks seid) dann gerne her damit, dann kann ich das in Richtung eQ3 tragen...

**Reply by stan23** [reply]
> Wenn in einer Minute der CS auf 100% steigt, sind 10% 6 Sekunden in denen das Funkmodul nicht sendet?! Und das reicht um das anlernen zu torpedieren?

Nein, das war keine belastbare Messung von mir sondern nur der Spielversuch um zu zeigen, dass der GDO2-Pin zum CS führt und wie der Zusammenhang ganz grob ist.

Was der HmIP-RFUSB genau macht, ob er also nur genau dann sendet wenn per GDO2 kein RSSI >-91.5 dBm gemessen wird, oder ob er da noch eine zeitliche Abhängigkeit reinbringt, wissen wir nicht und können es durch Blackbox-Messungen nur schwerlich rausfinden.
Den Entwicklern liegen diese Informationen hingegen offen ("die Wahrheit liegt im Code").

Messen können wir von außen welche Signalfolge an GDO2 zu welcher CS-Berechnung führt, und das nur mit mittlerem Aufwand und nicht völlig zweifelsfrei.

**Comment by Micha-DU** [comment]
Das kann man jetzt auch "gegenargumentieren": gerade weil die bisherigen Komponenten eher "teuer" waren, ist langsam mal ein bisschen sparen angesagt ;-) Die 4 BWTH, der Raspi plus Gehäuse usw. waren nicht zum Nulltarif zu haben ;-)

Auf der anderen Seite hast Du Recht: man merkt bei den HmIP-Komponenten meiner Meinung nach schon, dass sie in einer anderen Liga spielen als irgendwelche WiFi-Aktoren aus China. Trotzdem muss ja auch das Kosten-Nutzen-Verhältnis (wie immer) im Fokus bleiben. Und solange das "kleine" Modul meinen Anforderungen genügt, ist es durchaus eine Alternative. Ansonsten könnte man sich ja vielleicht auf eine Art "Mittelweg" einigen und sagen, dass ein neuer RF-USB kommt... hab ich gerade mal so überlegt.

Witzigerweise habe ich bei den Komponenten so absolut gar keine Probleme, was die Kommunikation, das Anlernen oder sonstiges angeht... bei all den Problemen, die unsere Heizungsanlage so mitbringt, ist das echt wohltuend ;-)

**Comment by Micha-DU** [comment]
Herrjee... also jetzt bin ich erstmal "raus" und verstehe nicht mehr so wirklich viel ;-)

Aber: Ja, ich habe mich nun für einen RF-USB entschieden, der ganz zufällig 5 km von mir entfernt auf einem Kleinanzeigenportal auftauchte. Kurz den Lötkolben geschwungen (hat mal wieder Spaß gemacht) und alles funktioniert wieder einwandfrei.

ABER: mir ist im Gegensatz zu meinem bisherigen USB-Stick etwas aufgefallen (ich weiß leider nicht, was das für ein Teil ist/war - jedenfalls wurde er nicht als "Gerät" in der CCU bzw. in der raspberrymatic geführt), was die "Performance" angeht.

Bisher war es so, dass die "Status"-/Startseite quasi augenblicklich geladen wurde, wenn ich mich angemeldet oder innerhalb einer Sitzung auf die Startseite gewechselt habe (dort sind 4 Thermostate hinterlegt und ich habe dazu noch den einen oder anderen Kanal anzeigen lassen, z. B. den Schaltzustand des Kanals 10 der BWTHs - diese Anzeige ist nun jedoch plötzlich "leer"). Momentan dauert es aber bis zu 3 Sekunden und ich sehe oben links nur das rotierende "Loading"-Symbol. Da ich aber fast zeitgleich auch das Update auf die 3.67.10.20230114 durchgeführt habe, weiß ich nun nicht, ob es an der neuen Version oder am Einsatz des RF-USB liegt.

Hat dazu vielleicht jemand eine Info oder Erfahrungswerte?

VG Micha
